redash/handlers/queries.py

```diff
@@ -171,6 +171,12 @@ def get(self):
 
         return response
 
+def require_access_to_dropdown_queries(user, query_def):
+    parameters = query_def.get('options', {}).get('parameters', [])
+    dropdown_queries = [models.Query.get_by_id(p["queryId"]) for p in parameters if p["type"] == "query"]
```

Comment:
It was actually using `query.in_` before (which is a single query), but it didn't fail on queries outside the range (and that led to a situation in which you could have specified a non-existing query id, which may exist later on). Given that most queries have a very small number dropdowns, I felt its best not to handle it and keep the code clean.

I guess I could find another way of writing it in a single query, I just didn't see it as that important.

Reply:
Good catch. We can verify that all the ids passed in were returned in the database response?

While this is generally not a big deal, these things tend to accumulate. Considering this will run on every query save, it can be an annoyance. If the fix isn't that complex, worth handling.

Also, please add a test case that verifies you can't pass in non existing query ids.

Reply:
b4fca74f does the trick. Let me know if you find it better or worse.